src/mam4xx/modal_aer_opt.hpp

```diff
@@ -964,32 +965,33 @@ void modal_aero_sw(const ThreadTeam &team, const Real dt,
 
   team.team_barrier();
 
-  for (int kk = top_lev; kk < pver; ++kk) {
+  for (int isw = 0; isw < nswbands; ++isw) {
+
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, pver), [&] (const int kk) {
 
-    for (int isw = 0; isw < nswbands; ++isw) {
       Kokkos::parallel_reduce(
-          Kokkos::TeamThreadRange(team, ntot_amode),
+          Kokkos::ThreadVectorRange(team, ntot_amode),
           [&](int imode, Real &suma) { suma += tauxar_work(kk, imode, isw); },
           tauxar(isw, kk + 1));
 
       Kokkos::parallel_reduce(
-          Kokkos::TeamThreadRange(team, ntot_amode),
+          Kokkos::ThreadVectorRange(team, ntot_amode),
           [&](int imode, Real &suma) { suma += wa_work(kk, imode, isw); },
           wa(isw, kk + 1));
 
       Kokkos::parallel_reduce(
-          Kokkos::TeamThreadRange(team, ntot_amode),
+          Kokkos::ThreadVectorRange(team, ntot_amode),
           [&](int imode, Real &suma) { suma += ga_work(kk, imode, isw); },
           ga(isw, kk + 1));
 
       Kokkos::parallel_reduce(
-          Kokkos::TeamThreadRange(team, ntot_amode),
+          Kokkos::ThreadVectorRange(team, ntot_amode),
           [&](int imode, Real &suma) { suma += fa_work(kk, imode, isw); },
           fa(isw, kk + 1));
```

Comment:
In EAMxx, the `compute-sanitizer --tool=racecheck --racecheck-detect-level=error` gave errors in these parallel_reduce calls, FYI.